src/mds/Server.cc

```diff
@@ -799,7 +799,9 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
     break;
 
   default:
-    ceph_abort();
+    derr << "Server received unknown message " << m->get_type() << ", closing session and blocklisting the client " << session->get_client() << dendl;
+    CachedStackStringStream css;
+    mds->evict_client(session->get_client().v, false, true, *css, nullptr);
```

Comment:
@batrick After some discussion with @lxbsz, maybe we should be evicting client here instead of harshly aborting MDS? Rationale is: When we reach `dispatch()`, it should be one of the 4 types types, if not then there is definitely something seriously buggy/fishy there, therefore we just directly abort the MDS with the statements in the commit 8e46a139b263fa3f53926ba1d83b3d220425b7f1, so MDS case is handled there while we handle the case of clients in `handle_client_session` in more better way than just abruptly aborting(in case of unknown messages) even without a single message. What do you think about it?

Reply:
Okay, thanks. I have fixed it but evicting the client and NOT aborting MDS.